subprojects/testing-base/src/integTest/groovy/org/gradle/testing/TestResultIntegrationTest.groovy

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.testing
+
+import org.gradle.api.internal.tasks.testing.report.HtmlTestResultsFixture
+import org.gradle.integtests.fixtures.AbstractIntegrationSpec
+import org.gradle.integtests.fixtures.JUnitXmlTestExecutionResult
+import org.gradle.integtests.fixtures.TestResources
+import org.gradle.test.fixtures.file.TestFile
+import org.junit.Rule
+import spock.lang.Issue
+
+class TestResultIntegrationTest extends AbstractIntegrationSpec {
+    @Rule
+    final TestResources resources = new TestResources(testDirectoryProvider)
+    final TestFile indexFile = testDirectoryProvider.file("build/reports/tests/test/index.html")
+
+    @Issue("https://github.com/gradle/gradle/issues/20081")
+    def "testDisabledParameterizedReport"() {
+        given:
+        executer.withRepositoryMirrors()
+        executer.withStackTraceChecksDisabled()
+
+        when:
+        run "test"
+
+        then:
+        def html = new HtmlTestResultsFixture(indexFile)
```

Comment:
Consider to use DefaultTestExecutionResult. It's contains html&xml results. It's instantiated with testDirectory and doesn't force you to keep reference index.html file. 
At glance it could make your tests less verbose and fit well to your purposes

Reply:
Thank you for the guide! updated